ui/src/dashboards/reducers/dashTimeV1.js

```diff
@@ -0,0 +1,44 @@
+const initialState = {
+  ranges: [],
+}
+
+const dashTimeV1 = (state = initialState, action) => {
+  switch (action.type) {
+    case 'ADD_DASHBOARD_TIME_V1': {
+      const {dashboardID, timeRange} = action.payload
+      const ranges = [...state.ranges, {dashboardID, timeRange}]
+
+      return {...state, ranges}
+    }
+
+    case 'DELETE_DASHBOARD': {
+      const {dashboardID} = action.payload
+      const ranges = state.ranges.filter(r => r.dashboardID !== dashboardID)
+
+      return {...state, ranges}
+    }
+
+    case 'SET_DASHBOARD_TIME_V1': {
+      const {dashboardID, timeRange} = action.payload
+      const exists = state.ranges.find(r => r.dashboardID === dashboardID)
+      const {upper, lower} = timeRange
```

Comment:
I believe this function and be simplifed to a three liner:
https://lodash.com/docs/4.17.4#unionWith